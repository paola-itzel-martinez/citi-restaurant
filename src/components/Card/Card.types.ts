import { HTMLAttributes } from "react";

export interface CardCSS extends HTMLAttributes<HTMLDivElement> {}

export interface CardProps extends CardCSS {
  children: React.ReactNode;
}
