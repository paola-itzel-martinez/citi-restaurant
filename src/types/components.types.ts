export enum ButtonVariant {
  CARD = 'CARD',
  MENU = 'MENU',
}

export enum DividerVariant {
  HORIZONTAL = 'HORIZONTAL',
  VERTICAL = 'VERTICAL',
}

export enum Portals {
  MODAL = 'MODAL',
}

export enum TypographyVariant {
  BUTTON = 'BUTTON',
  BUTTON_MENU = 'BUTTON_MENU',
  NORMAL = 'NORMAL',
  SMALL = 'SMALL',
  TITLE = 'TITLE',
}

export enum TypographyFamily {
  GENERAL = "'Roboto',sans-serif",
}

export type CSSRule = number | string;

export type CSSMarginRules = {
  m?: CSSRule;
  mx?: CSSRule;
  my?: CSSRule;
  mt?: CSSRule;
  mr?: CSSRule;
  mb?: CSSRule;
  ml?: CSSRule;
};

export type CSSPaddingRules = {
  p?: CSSRule;
  px?: CSSRule;
  py?: CSSRule;
  pt?: CSSRule;
  pr?: CSSRule;
  pb?: CSSRule;
  pl?: CSSRule;
};

export type CSSSpacingRules = {
  width?: CSSRule;
  minWidth?: CSSRule;
  maxWidth?: CSSRule;
  height?: CSSRule;
  minHeight?: CSSRule;
  maxHeight?: CSSRule;
};
