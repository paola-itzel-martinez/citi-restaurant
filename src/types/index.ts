export * from './components.types';
export * from './mediaQueries.types';
export * from './yelpAPI.types';
