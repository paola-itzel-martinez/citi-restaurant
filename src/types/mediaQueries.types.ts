export enum SCREEN_SIZE {
  DESKTOP = 'DESKTOP',
  TABLET = 'TABLET',
  MOBILE = 'MOBILE',
}
