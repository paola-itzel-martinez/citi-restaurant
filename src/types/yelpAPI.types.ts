export type Category = {
  title: string;
  alias: string;
};

export type Business = {
  id: string;
  categories: Category[];
  image_url: string;
  name: string;
  rating: number;
  url: string;
  price?: string;
};
