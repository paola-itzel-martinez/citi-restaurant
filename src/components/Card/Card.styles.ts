import styled from "styled-components";
import { MEDIA_MOBILE, palette, shadow } from "../../styled";
import { CardCSS } from "./Card.types";

export const Wrapper = styled.div<CardCSS>`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 28px;
  gap: 16px;
  background: ${palette.WHITE};
  box-shadow: ${shadow.NORMAL};
  border-radius: 8px;

  &:hover {
    cursor: ${({ onClick }) => (onClick ? "pointer" : "default")};
  }

  @media ${MEDIA_MOBILE} {
    box-shadow: unset;
    padding: 0;
    width: 100%;
  }
`;
