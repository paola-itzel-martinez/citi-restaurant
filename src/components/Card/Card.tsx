import React from "react";
import { CardProps } from "./Card.types";
import { Wrapper } from "./Card.styles";

const Card = (props: CardProps, ref: React.Ref<HTMLDivElement>) => {
  const { children, ...otherProps } = props;

  return (
    <Wrapper ref={ref} {...otherProps}>
      {children}
    </Wrapper>
  );
};

export default React.forwardRef(Card);
