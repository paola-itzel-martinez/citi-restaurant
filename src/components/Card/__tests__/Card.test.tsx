import React from "react";
import { customRender, screen } from "../../../../tests";
import { CardProps } from "../Card.types";
import { Card } from "..";

const CHILDREN = "Children";

const GeneralComponent = (props: Omit<CardProps, "children">) => (
  <Card {...props}>{CHILDREN}</Card>
);

describe("<Card />", () => {
  it("Should have a default css rules by default", () => {
    const { container } = customRender(<GeneralComponent />);

    expect(screen.getByText(CHILDREN)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
