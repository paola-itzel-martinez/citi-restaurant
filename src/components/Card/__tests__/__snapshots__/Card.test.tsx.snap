// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Should have a default css rules by default 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 28px;
  gap: 16px;
  background: #ffffff;
  box-shadow: 0px 0.25px 1px rgba(0,0,0,0.039),0px 0.85px 3px rgba(0,0,0,0.19);
  border-radius: 8px;
}

.c0:hover {
  cursor: default;
}

@media (min-width:0px) and (max-width:599px) {
  .c0 {
    box-shadow: unset;
    padding: 0;
    width: 100%;
  }
}

<div>
  <div
    class="c0"
  >
    Children
  </div>
</div>
`;
